src/java/org/apache/cassandra/locator/ReplicaPlans.java

```diff
@@ -610,8 +613,11 @@ public static ReplicaPlan.ForTokenRead forRead(Keyspace keyspace, Token token, I
     public static ReplicaPlan.ForRangeRead forRangeRead(Keyspace keyspace, Index.QueryPlan indexQueryPlan, ConsistencyLevel consistencyLevel, AbstractBounds<PartitionPosition> range, int vnodeCount)
     {
         AbstractReplicationStrategy replicationStrategy = keyspace.getReplicationStrategy();
-        EndpointsForRange candidates = candidatesForRead(keyspace, indexQueryPlan, consistencyLevel, ReplicaLayout.forRangeReadLiveSorted(replicationStrategy, range).natural());
-        EndpointsForRange contacts = contactForRead(replicationStrategy, consistencyLevel, false, candidates);
+        IEndpointSnitch endpointSnitch = DatabaseDescriptor.getEndpointSnitch();
+        EndpointsForRange candidates = candidatesForRead(keyspace, indexQueryPlan, consistencyLevel, ReplicaLayout.forRangeReadLiveSorted(replicationStrategy, range).natural())
+                                       .filter(endpointSnitch.filterByAffinity(keyspace.getName()));
+        EndpointsForRange contacts = contactForRead(replicationStrategy, consistencyLevel, false, candidates)
+                                     .filter(endpointSnitch.filterByAffinity(keyspace.getName()));
```

Comment:
Basically the same as above - this is the best place I could find to introduce `filterByAffinity`.

The only use of `ReplicaPlans.forRangeRead` is in `ReplicaPlanIterator.computeNext` and this corresponds to the change in DSE for `filterByAffinity` in `RangeIterator.computeNext`.